### PR TITLE
Feat(new browser tab investigations): choosing investigation in landing tab now opens new tab in browser for investigation

### PR DESCRIPTION
**Pull request by ShiranEyal**

### CODE REVIEW

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 1]
```diff
@@ -100,8 +102,8 @@ const useInvestigationTable = (): useInvestigationTableOutcome => {
   }, [user.id, classes.errorAlertTitle]);
 
   const moveToTheInvestigationForm = (epidemiologyNumberVal: number) => {
-    setEpidemiologyNum(epidemiologyNumberVal);
-    history.push('/investigation');
+    localStorage.setItem("selectedEpidemiologyNumber", JSON.stringify(epidemiologyNumberVal));
```
Comment (sandymash) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:105]: local storage is very vaulnerable.
I would try using useHistory with state

Comment (sandymash) [reply client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:105]: if that doesnt work maybe url params/query?

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx [hunk 2]
```diff
@@ -55,7 +56,31 @@ const InvestigationInfoBar: React.FC<Props> = ({ onExitInvestigation }: Props) =
...
+            },
+            timer: 1750,
+            showConfirmButton: false
+        });
+
+        timeout(1900).then(() => history.push(landingPageRoute));
+    }
+
+    React.useEffect(() => {
+        if (localStorage.selectedEpidemiologyNumber) {
```
Comment (OrelKaravani) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx:74]: Why are you using localStorage?

**Review by OrelKaravani — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx (hunk 2 above)
Comment (ShiranEyal) [reply client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx:74]: i need a way to send data between my tabs when i choose an investigation, this seemed like the best way. its either this or passing the epidemiology number via url which seemed to me more vulnerable

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx [hunk 3]
```diff
@@ -55,7 +56,30 @@ const InvestigationInfoBar: React.FC<Props> = ({ onExitInvestigation }: Props) =
 
     const epidemiologyNumber = useSelector<StoreStateType, number>(state => state.investigation.epidemiologyNumber);
 
-    React.useEffect(() => { 
+    let { selectedEpidemiologyNumber } = useParams();
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx:59]: Const

File: client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/usePersonalInfoTab.ts [hunk 4]
```diff
@@ -14,7 +14,7 @@ const usePersonalInfoTab = (parameters: usePersoanlInfoTabParameters): usePerson
     const {setOccupations, setInsuranceCompanies, personalInfoStateContext, 
         setSubOccupations, setSubOccupationName, setCityName, setStreetName, setStreets} = parameters;
 
-    const fetchPersonalInfo = () => {
+    const fetchPersonalInfo = (epidemiologyNumber: number) => {
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/usePersonalInfoTab.ts:17]: Where do you use this parameter?

**Review by yaniryos — APPROVED** [review]

**Review by daniel7764 — DISMISSED** [review]

File: client/src/Utils/axios.ts [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import axios from 'axios';
 
 import { store } from 'redux/store';
+import StoreStateType from 'redux/storeStateType';
```
Comment (eladyashar) [inline client/src/Utils/axios.ts:4]: not in use in this file - remove import

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx [hunk 6]
```diff
@@ -57,7 +59,34 @@ const InvestigationInfoBar: React.FC<Props> = ({ currentTab }: Props) => {
...
+            },
+            timer: 1750,
+            showConfirmButton: false
+        });
+
+        timeout(1900).then(() => history.push(landingPageRoute));
+    }
+
+    React.useEffect(() => {
+        timeout(2000).then(() => {
```
Comment (eladyashar) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx:77]: why timeout is needed here ?

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx [hunk 7]
```diff
@@ -57,7 +59,34 @@ const InvestigationInfoBar: React.FC<Props> = ({ currentTab }: Props) => {
...
+            let openedEpidemiologyNumber = store.getState().investigation.lastOpenedEpidemiologyNumber;
+            if (openedEpidemiologyNumber !== -1) {
+                setEpidemiologyNum(openedEpidemiologyNumber);
+                setLastOpenedEpidemiologyNum(-1);
```
Comment (sandymash) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx:81]: maybe store default user as some global const, or just save NULL?

Comment (sandymash) [reply client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx:81]: I mean the -1

**Review by yaniryos — DISMISSED** [review]

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx (hunk 6 above)
Comment (ShiranEyal) [reply client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx:77]: the package that i used (redux state sync) updates each tab when opened, depending on the previous tabs state. to do so it dispatches certain actions after the tab has been loaded. this timeout is to ensure that the state sync has finished before checking which investigation was chosen or if there is need to go back to landing page. if you think you have a better solution im more than happy to hear it.

**Review by eladyashar — APPROVED** [review]

**Review by daniel7764 — APPROVED** [review]

**Pull request merged**